data/transactions/logic/assembler.go

```diff
@@ -1541,73 +1563,78 @@ func (ops *OpStream) assemble(text string) error {
 	for scanner.Scan() {
 		ops.sourceLine++
 		line := scanner.Text()
-		line = strings.TrimSpace(line)
-		if len(line) == 0 {
-			ops.trace("%3d: 0 line\n", ops.sourceLine)
-			continue
-		}
 		if strings.HasPrefix(line, "//") {
-			ops.trace("%3d: // line\n", ops.sourceLine)
-			continue
-		}
-		if strings.HasPrefix(line, "#pragma") {
-			ops.trace("%3d: #pragma line\n", ops.sourceLine)
-			ops.pragma(line)
+			ops.trace("%3d: comment\n", ops.sourceLine)
 			continue
 		}
 		fields := fieldsFromLine(line)
-		if len(fields) == 0 {
-			ops.trace("%3d: no fields\n", ops.sourceLine)
-			continue
-		}
-		// we're about to begin processing opcodes, so settle the Version
-		if ops.Version == assemblerNoVersion {
-			ops.Version = AssemblerDefaultVersion
-		}
-		if ops.versionedPseudoOps == nil {
-			ops.versionedPseudoOps = prepareVersionedPseudoTable(ops.Version)
-		}
-		opstring := fields[0]
-		if opstring[len(opstring)-1] == ':' {
-			ops.createLabel(opstring[:len(opstring)-1])
-			fields = fields[1:]
-			if len(fields) == 0 {
-				ops.trace("%3d: label only\n", ops.sourceLine)
+		for current, next := processFields(fields); len(current) > 0 || len(next) > 0; current, next = processFields(next) {
+			if len(current) == 0 {
 				continue
 			}
-			opstring = fields[0]
-		}
-		spec, expandedName, ok := getSpec(ops, opstring, fields[1:])
-		if ok {
-			ops.trace("%3d: %s\t", ops.sourceLine, opstring)
-			ops.recordSourceLine()
-			if spec.Modes == modeApp {
-				ops.HasStatefulOps = true
+			opstring := current[0]
+			if opstring == "#pragma" {
+				ops.trace("%3d: #pragma line\n", ops.sourceLine)
+				// pragma get the rest of the tokens
+				pragmaFields := current[:cap(current)]
```

Comment:
This seems real iffy. Yes, I suppose we know that because of the way `processFields` works, this will recover the original line. But man, it's subtle. There's certainly nothing in the spec of `processFields` that ensures this will work.

Reply:
Yaaaaa, it just felt so perfect to do this, especially since macros would work with this as well. I'll change it to use appends.

Reply:
Actually, I just realized we have no way of knowing how many semicolons they used between current and next without doing this or going back to the line and getting fields from line (in which case we have to find the index of current), so this may be the best way

Reply:
I'm pushing a change, but I'll note that it's sort of a fake b/c we end up not being able to know if there is a terminating semicolon if next is empty. Also, I realized we are definitely at least going to want define to be on its own line b/c we only want the macro body to be expanded and not the macro name (i.e. "#define x 1\n #define x 3" should not expand to "#define x 1\n #define 3 1")